sky/server/server.py

```diff
@@ -1749,6 +1749,9 @@ async def root():
                                                     cmd_args.metrics_port),
                                               daemon=True)
             metrics_thread.start()
+        gc_thread = threading.Thread(target=requests_lib.requests_gc_daemon,
+                                     daemon=True)
+        gc_thread.start()
```

Comment:
wondering if it would be more efficient to use asyncio instead of thread for GC and metrics above.